### PR TITLE
Add docs on download stats

### LINKED ISSUE #100: How are download stats generated for datasets on the hub
I am hoping this is the most appropriate repository for this topic. Let me know if I should open elsewhere. I have tried searching for this information but I may have missed something.

**Is your feature request related to a problem? Please describe.**
The Hugging Face hub provides some download stats in the public interface for datasets: 
Screenshot 2022-04-14 at 16 03 43

It would be useful to know a bit more about how they are calculated in different situations:

### For datasets hosted outside the hub (i.e. a dataset script points to files hosted somewhere else):

I'm assuming, in this case, a download is registered when you do: 

```python
ds = load_dataset('awesome-dataset')
```

### For datasets hosted inside the hub (i.e. a dataset script points to files hosted somewhere else).

In the case where files are hosted directly on the hub, is there any distinction made between the number of files that are downloaded? i.e. if a subset of a dataset is loaded i.e. 

```python
ds = load_dataset('awesome-dataset', 'subset')
```
and this only requires accessing a subset of files hosted on the hub is this counted differently from downloading all the data hosted on the hub?

### For direct downloads of files hosted on the hub. 

For example, if for the dataset hosted at https://huggingface.co/datasets/w11wo/imdb-javanese I `wget` the underlying `csv.zip` files, is that registered in the download stats? 

**Describe the solution you'd like**
It would be great to provide more documentation (apologies if I've missed existing documentation) on how the download stats are generated for datasets.

For context in the library/academic publishing word world, some repositories/publishers follow [COUNTER](https://www.projectcounter.org/) guidance on recording stats. For example, repositories that follow this: https://www.projectcounter.org/code-practice-research-data/repositories-that-have-implemented-the-code-of-practice-for-research-data/. This is intended to ensure that you can make more valid comparisons across platforms. This is likely to be relevant only to a subset of your users so might not be worth the effort to meet these requirements but adding documentation about how stats are generated might be helpful. There is an implementation guide related to this: https://github.com/CDLUC3/Make-Data-Count/blob/master/getting-started.md 

**Describe alternatives you've considered**
For files hosted outside of Hugging Face, it is often possible to get download stats however, it would be useful to be able to cross-reference those with the hub stats. 

**Additional context**
Having download stats is useful in a few scenarios:

- For institutions thinking about sharing datasets, they often need to demonstrate impact. One way to do this is to cite download stats from the Hub. In particular, it is useful to be able to compare those with other stats in order to have a sense of how much putting data in the Hub helps with overall usage. As an example, the British Library recently added some datasets to the hub (https://huggingface.co/datasets/blbooksgenre) -- it would be great to know how much doing this has generated new usage compared to only hosting the files on the BL's data repository. 

- For researchers/academics, there is a slow move away from only focusing on 'traditional' metrics i.e. citation counts, h-index etc. to also considering other ways in which they can make contributions. Again, contributing a dataset to the hub could be a valuable contribution that they may want to cite in whatever reporting structures they have in their university/country/funder -- for this having more confidence in the stats from the hub could be very helpful in demonstrating impact.

### PR DESCRIPTION
**Pull request by osanseviero**
Fix #100

### CODE REVIEW

File: docs/hub/models-faq.md
Comment (osanseviero) [reply docs/hub/models-faq.md]: Done! I was thinking in having all libraries listed here, but it would be a bit painful to maintain I think, so we can keep this as is

File: docs/hub/models-faq.md
Comment (julien-c) [inline docs/hub/models-faq.md:47]: let's define an anchor so that we can link directly to here (w/o risking breaking it with a header change)

Or even have this section as its own doc?

File: docs/hub/models-faq.md
Comment (julien-c) [inline docs/hub/models-faq.md:49]:
```suggestion
Counting the number of downloads for models is not a trivial task as a single model repository might contain multiple files, including multiple model weight files (e.g., with sharded models), and different formats depending on the library. To avoid double counting downloads (e.g., counting a single download of a model as multiple downloads), the Hub uses a set of query files that are employed for download counting. 
```

Comment (julien-c) [reply docs/hub/models-faq.md:49]: maybe give an example of ES mapping (or maybe it's too technical/detailed?)

File: docs/hub/models-faq.md
Comment (julien-c) [inline docs/hub/models-faq.md:51]:
```suggestion
Every HTTP request to these files, including `GET` and `HEAD` will be counted as a download. By default, when no library is specified, the Hub uses `config.json` as the default query file. Otherwise, the query file depends on each library, and the Hub might examine files such as `pytorch_model.bin` and `adapter_config.json`.
```

File: docs/hub/datasets-faq.md
Comment (julien-c) [inline docs/hub/datasets-faq.md:8]:
```suggestion
The Hub provides download stats for all datasets loadable via the `datasets` library. The system is quite different from how [model downloads](./link to page) are counted

To determine the number of downloads, the Hub counts every time `load_dataset` is called in Python, excluding Hugging Face's CI tooling on GitHub. This means that:

* Whether the data is directly stored on the Hub repo or if the repository has a script to load the data from an external source, the download count is not impacted.
* If a user manually downloads the data using tools like `wget` or through the Hub's user interface (UI), those downloads will not be included in the download count.
```

Comment (julien-c) [reply docs/hub/datasets-faq.md:8]: note that this section is accurate now, but it's interesting to think how it could evolve if we move to support more data libraries in the future

**Review by julien-c — COMMENTED** [review]
left a few comments

File: docs/hub/datasets-faq.md
Comment (davanstrien) [inline docs/hub/datasets-faq.md:7]:
```suggestion
* The download count is the same regardless of whether the data is directly stored on the Hub repo or if the repository has a script to load the data from an external source.
```

File: docs/hub/datasets-faq.md
Comment (davanstrien) [inline docs/hub/datasets-faq.md:8]:
```suggestion
* If a user manually downloads the data using tools like `wget` or the Hub's user interface (UI), those downloads will not be included in the download count.
```

**Review by davanstrien — APPROVED** [review]
A few small suggestions (feel free to ignore!)

**Comment by Wauplin** [comment]
(can we wait Monday before merging even if already approved? Just want to read it without rushing and won't have time today :) )

**Comment by julien-c** [comment]
yes for sure let's wait a few days – also interested in @LysandreJik's read

File: docs/hub/datasets-download-stats.md
Comment (LysandreJik) [inline docs/hub/datasets-download-stats.md:5]: I would maybe clarify that nothing is sent from the user's setup, it's just the server receiving a request to return a file that we count. 

Maybe a line such as 
> No information is sent from the user and no additional calls are made for this. The count is done server-side as we serve files for downloads.

**Review by LysandreJik — APPROVED** [review]
Looks good to me in principle, but I'd add one or two lines to specify explicitly that we're not doing anything funky on the user's side. We're absolutely optimizing for minimal number of requests and we're not sending any information. Everything happens server-side.

File: docs/hub/datasets-download-stats.md
Comment (julien-c) [reply docs/hub/datasets-download-stats.md]: yes great point @LysandreJik

**Review by Wauplin — APPROVED** [review]
Looks good!

**Comment by osanseviero** [comment]
Thanks everyone for the reviews. I'll merge this later today if nobody else adds comments

**Review by pcuenca — APPROVED** [review]
Looks great!

A small clarification regarding model downloads would be to define a "download" as every use of the target file, including when it's already cached in the user's computer. Probably not necessary.

**Comment by osanseviero** [comment]
For now let's go with this and later we can iterate

**Pull request merged**
